src/ui/EditorNS/jsproxy.cpp

```diff
@@ -0,0 +1,116 @@
+#include "include/EditorNS/jsproxy.h"
+#include <QDebug>
+
+namespace EditorNS {
+
+QVariant JsToCppProxy::getResult()
+{
+    return m_result;
+}
+
+QVariant JsToCppProxy::getCursor()
+{
+    return m_cursor;
+}
+
```

Comment:
I'm not a fan of these hardcoded methods. Can't we use something like a generic map instead of multiple fields like these?

So that we can remove all these methods from here and move them in Editor, like this (pseudocode):

```
Editor::getResult()
 {
     return proxy->values.get("result");
 }
 
Editor::getCursor()
 {
     return proxy->values.get("cursor");
 }

Editor::getField(string field)
 {
    return proxy->values.get(field);
 }
```

That's important for future extensibility with extensions.